### ISSUE TITLE
[ISSUE/QUESTION] Matchbot Status

### ISSUE THREAD

**Issue by yuyiken** [SmileYzn/MatchBot#54]
Hi, I was testing it for a while and got in this issue. 

For example I'm playing a match with bots and when I leave, the match is still running. 

All bots are kicked, and no one is on server, as u can see it still counting rounds. 

![1](https://user-images.githubusercontent.com/74840575/235686471-b80c9e9f-b8e4-41a8-b806-7d383711b928.jpg)

When I join again game continues as was before buy in my scoreboard are 0-0. 

If I win or lose rounds game for example is going 4-5 and I have 0-1. 

Another issue I have, got it twice only. When I do this and leave a server for night. 

When start game next day. It do it starts like 1-0 not 0-0. 

I got this issue only twice when game starts 1-0 not 0-0.  

I think it should be something relationed with issue before.

**Comment by SmileYzn** [comment]
> 

Did you make sure that BOTS leaves the game?
There i need to use bot_quota 0 after disconnects the game.

And yes, the games was set to ended state.

**Comment by yuyiken** [comment]
@SmileYzn no, bot_quota still in 9. There is a problem then. I only appreciate that they are kicked, but bot_quota still in 9

**Comment by SmileYzn** [comment]
> @SmileYzn no, bot_quota still in 9. There is a problem then. I only appreciate that they are kicked, but bot_quota still in 9. Anyways server was empty, If any bot is connected it appears in left side of server browser.

Did you do a status command after you disconnect from server?
MatchBot only consider a empty server if player and bots is not playing.

There is a problem with bots variables, not matchbot itself 🤔

**Comment by SmileYzn** [comment]
Did worked? I really did not understand.

**Comment by yuyiken** [comment]
> 
Hi, no that I tried to explain that it not worked for me.

**Comment by SmileYzn** [comment]
@yuyiken 

Try to see if was fixed here: https://github.com/SmileYzn/MatchBot/actions/runs/4889671823

I forgot to put what match bot does when server gets empty while vote session is running.
In that case, i do a restart map since server can be empty too.

**Comment by yuyiken** [comment]
@SmileYzn Hi, I was testing and get same results, when the game is live and bot are disconected from game or I leave and bots are kicked. Nothing changes, game still live until half time with empty server.

**Comment by SmileYzn** [comment]
@yuyiken 
Did you put mb_vote_map in matchbot.cfg?

I have sure that issue is fixed here:

```
void CMatchAdminMenu::MapMenuHandle(int EntityIndex, P_MENU_ITEM Item)
{
	auto Player = UTIL_PlayerByIndexSafe(EntityIndex);

	if (Player)
	{
		gMatchBot.m_VoteMap->value = 0.0f;

		gMatchChangeMap.ChangeMap(Item.Text, 5.0f, true);

		gMatchUtil.SayText(nullptr, Player->entindex(), _T("\3%s\1 changed map to: \4%s\1..."), STRING(Player->edict()->v.netname), Item.Text.c_str());
	}
}

Ps. You can talk in Spanish here. I can understand too.
```

**Comment by SmileYzn** [comment]
@yuyiken 

Try to fix it here: https://github.com/SmileYzn/MatchBot/actions/runs/4898876553

I think that function is counting players always return one player since player is not fully disconnected from server yet.

**Comment by yuyiken** [comment]
> @yuyiken Did you put mb_vote_map in matchbot.cfg?
> 
> I have sure that issue is fixed here:
> 
> ```
> void CMatchAdminMenu::MapMenuHandle(int EntityIndex, P_MENU_ITEM Item)
> {
> 	auto Player = UTIL_PlayerByIndexSafe(EntityIndex);
> 
> 	if (Player)
> 	{
> 		gMatchBot.m_VoteMap->value = 0.0f;
> 
> 		gMatchChangeMap.ChangeMap(Item.Text, 5.0f, true);
> 
> 		gMatchUtil.SayText(nullptr, Player->entindex(), _T("\3%s\1 changed map to: \4%s\1..."), STRING(Player->edict()->v.netname), Item.Text.c_str());
> 	}
> }
> ```
> 
> Ps. You can talk in Spanish here. I can understand too.

You are right. I put it in matchbot.cfg and works fine. It was my mistake.
But I tried to take value of this cvar without putting into matchbot and get value of 0 when game is live. 

> @yuyiken
> 
> Try to fix it here: https://github.com/SmileYzn/MatchBot/actions/runs/4898876553
> 
> I think that function is counting players always return one player since player is not fully disconnected from server yet.

So I tested and still have same issue. I really don't understand how can be possible that player is not fully disconnected. If he drops, time out or get kicked.. If playerdisconects function not working properly you can do the check when round ends or starts.. and game is live.. But counting players should work fine since you are using it for ready timer and another stuff.. 

PS: I prefer write in english because of people with issues that read in future can solve their issues faster..

**Comment by SmileYzn** [comment]
> and still have same issue. I really don't understand how can be possible that player is not fully disconnected. If he drops, time out or get kicked.. If playerdisconects function not working properly you can do the check when round ends or starts.. and game is live.. But counting players should work fine since you are using it for ready t

mb_vote_map can't be put in any configuration file, it is used internally by Match BOT.
I have sure that is fixed when i disconnects from server or server was empty the match stops.

Did you can put an cvarlist of bot_ variables? used?

**Comment by yuyiken** [comment]
I only use 1 cvar, when I join I type bot_quota 9 when I want to test stop i put bot_quota 0

**Comment by yuyiken** [comment]
Hi, issue is fixed, I notice that it can be my problem with server problem. Please keep always server updated with correct rehlds version and regamedll_cs.
